src/gg.regression.unity.bots/Editor/Scripts/CodeGenerators/GenerateRGActionsClass.cs

```diff
@@ -0,0 +1,1024 @@
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+#if UNITY_EDITOR
+using System;
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.CSharp;
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+using static Microsoft.CodeAnalysis.CSharp.SyntaxFactory;
+#endif
+
+namespace RegressionGames.Editor.CodeGenerators
+{
+#if UNITY_EDITOR
+    // Dev Note: Not perfect, but mega time saver for generating this gook: https://roslynquoter.azurewebsites.net/
+    public static class GenerateRGActionsClass
+    {
+        public static Task Generate(
+            string filePath,
+            string behaviourName,
+            string behaviourNamespace,
+            List<RGActionAttributeInfo> botActions)
+        {
+
+            HashSet<string> usingSet = new()
+            {
+                "System",
+                "System.Collections.Generic",
+                "System.Collections.ObjectModel",
+                "Newtonsoft.Json",
+                "RegressionGames",
+                "RegressionGames.StateActionTypes",
+                "UnityEngine"
+            };
+
+            var className = $"RGActions_{behaviourName}";
+
+            var entityTypeName = behaviourName;
+            
+            List<SyntaxNodeOrToken> delegateList = new();
+            
+            List<MemberDeclarationSyntax> actionClassDeclarations = new();
+            // process each action
+            foreach (var botAction in botActions)
+            {
+                if (!botAction.ShouldGenerateCSFile)
+                {
+                    continue;
+                }
+                
+                entityTypeName = botAction.EntityTypeName ?? entityTypeName;
+                
+                // Add RGActionRequest class
+                actionClassDeclarations.Add(
```

Comment:
Is there a preferred way to review this code? It's pretty tricky to follow so I've mostly skimmed and assumed the structure was sound.

Reply:
Resolved in DMs